src/Language/PureScript/AST/Declarations.hs

```diff
@@ -481,14 +481,23 @@ pattern ValueDecl :: SourceAnn -> Ident -> NameKind -> [Binder] -> [GuardedExpr]
 pattern ValueDecl sann ident name binders expr
   = ValueDeclaration (ValueDeclarationData sann ident name binders expr)
 
+data DataConstructorDeclaration = DataConstructorDeclaration
+  { dataCtorAnn :: !SourceAnn
+  , dataCtorProperName :: !(ProperName 'ConstructorName)
+  , dataCtorVars :: ![(Ident, SourceType)]
```

Comment:
Did you call these by these names so as not to clash with the names of the same fields in the CST? I think it would make more sense to use the same names as in CST, i.e. `dataCtorName` and `dataCtorFields`, and then use qualification where both are in scope. (Presumably that's only `CST.Convert`?)